### ISSUE TITLE
"Particle Normalized Age" node not work on Particle Function

### ISSUE THREAD

**Issue by RuanLucasGD** [FlaxEngine/FlaxEngine#1640]
**Issue description:**

Using simple particle color logic:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/3ec2240b-93ac-4727-b1ba-cbfdc91e36cd)

Using particle function:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/38d97312-6698-49ed-9094-e771f2356155)
![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/94dc4d96-b9ae-416a-9248-a041b0577487)

**Flax version:**
1.6 master

**Comment by RuanLucasGD** [comment]
This issue has another problem, if you set emitter Simulation Mode as GPU using the Function, you cannot save your emitter file and Output show this error:

[ 00:02:50.397 ]: [Error] Failed to compile '29ab2b6bc46444a7ada94a85c1d6f61d'. C:\Users\ruanl\OneDrive\Documentos\Flax Projects\TestParticle\29ab2b6bc46444a7ada94a85c1d6f61d(175,18-23): error X3004: undeclared identifier 'local1'
```
[ 00:02:50.397 ]: [Error] Failed to compile 'CS_Main'
[ 00:02:50.398 ]: [Error] Failed to compile shader 'FlaxEngine.ParticleEmitter, 29ab2b6b44a7c464854aa9ad1df6d6c1, C:\Users/ruanl/AppData/Local/Temp/fa80a03c-47e5-cb67-3f0a-e1a9a6fb7bcd/29ab2b6bc46444a7ada94a85c1d6f61d.flax'
[ 00:02:50.398 ]: [Error] Cannot load 'FlaxEngine.ParticleEmitter, 29ab2b6b44a7c464854aa9ad1df6d6c1, C:\Users/ruanl/AppData/Local/Temp/fa80a03c-47e5-cb67-3f0a-e1a9a6fb7bcd/29ab2b6bc46444a7ada94a85c1d6f61d.flax' shader cache.
[ 00:02:50.398 ]: [Error] Loading asset 'FlaxEngine.ParticleEmitter, 29ab2b6b44a7c464854aa9ad1df6d6c1, C:\Users/ruanl/AppData/Local/Temp/fa80a03c-47e5-cb67-3f0a-e1a9a6fb7bcd/29ab2b6bc46444a7ada94a85c1d6f61d.flax' result: Failed.
[ 00:02:50.398 ]: [Warning] 'Content Load Task LoadAsset (4)' failed with result: AssetLoadError
```

**Comment by mafiesto4** [comment]
The problem was that the Particle Emitter Function had own particle data layout created for the graph and used it to sample particle data which was incorrect. I added a remapping table in 5b5f43714e36c685d346bdea437f8de9f5950664 which fixed CPU-particles.

For GPU particles I fixed that in d52412c5c3e00b4d8a185b56e82b52dd9b66ca71 and 2414abf1838fa2a97c84f9bc43c6142a25a187a6.

Also, if particle emitter function used any custom particle attributes they were missing but it's fixed in 8af2f458d7922a4a7780079cf37260b2027063ad.
